### PR TITLE
[TTS] Fix bugs in MagpieTTS evaluation

### PR DESCRIPTION
**Pull request by rlangman**
# What does this PR do ?

Fix three bugs in MagpieTTS evaluation and validation logic

**Collection**: TTS

# Changelog

- Save `utmosv2` values during evaluation so that they show up in the final evaluation plots
- Fix `_get_context_embeddings` so that after it pads audio and text context tensors to be the same length, it removes extra padding if the output only contains the shorter context type.
- Avoid logging context audio when text context is used. The current logic avoids this if all items in a batch are text context, but does not work when the batch contains a mix and text and audio context examples (which happens during our multi-lingual training). Logging empty audio throws a runtime error if tensorboard logger is enabled.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

### CODE REVIEW

File: nemo/collections/tts/models/magpietts.py [hunk 1]
```diff
@@ -1803,7 +1804,7 @@ def log_val_audio_example(
 
                 if is_wandb:
                     wandb_audio_log[f"Audio/Example_{idx}"] = list()
-                    if context_audio_np is not None:
+                    if context_audio_np is not None and context_audio_np.shape[0] > 0:
```
Comment (XuesongYang) [inline nemo/collections/tts/models/magpietts.py:1807]: logging only happens to the first batch at global rank 0. Let's say 1 of 3 examples only has text context, then `context_audio_np` should be `None`, then `context_audio_np` would not have shape. The current test condition seems already covering the scenario.

Am i missing something?

Comment (XuesongYang) [reply nemo/collections/tts/models/magpietts.py:1807]: again, we can simply the changes as below to cover both tests for `None` and/or empty numpy array. 
```suggestion
                    if not context_audio_np:
```

Comment (rlangman) [reply nemo/collections/tts/models/magpietts.py:1807]: Lets say a batch has 3 examples, the first having text conditioning and the other having audio conditioning. `context_audio_np` is not None, and `context_audio_lens[0] = 0`. `context_audio_np = context_audio_np[: context_audio_lens[0]]` will be an empty array, which is when the runtime error happens.

Changing it to  `if not context_audio_np:` also works, but numpy logs a warning saying you are not supposed to do that:

`DeprecationWarning: The truth value of an empty array is ambiguous. Returning False, but in future this will result in an error. Use array.size > 0 to check that an array is not empty.`
